mobile/library/common/internal_engine.cc

```diff
@@ -171,6 +172,15 @@ envoy_status_t InternalEngine::main(std::shared_ptr<Envoy::OptionsImplBase> opti
                   {Network::Address::IpVersion::v6});
             }
           }
+          if (Runtime::runtimeFeatureEnabled(
+                  "envoy.reloadable_features.reset_brokenness_on_nework_change")) {
+            Http::HttpServerPropertiesCacheManager& cache_manager =
+                server_->httpServerPropertiesCacheManager();
+
+            Http::HttpServerPropertiesCacheManager::CacheFn clear_brokenness =
+                [](Http::HttpServerPropertiesCache& cache) { cache.resetBrokenness(); };
+            cache_manager.forEachThreadLocalCache(clear_brokenness);
+          }
```

Comment:
I think I'm missing something here, but what causes this code to be run on network change (as opposed to running on startup or something else. I think I'm a bit lost in the woods, unfortunately)

Reply:
This would only apply if the alt-svc cache was persisted and we reloaded it on startup, right?  Don't we also need to add this call to https://github.com/envoyproxy/envoy/blob/7b8132ac125d9bd64089d65e3c5396e7aab2fde9/mobile/library/common/internal_engine.cc#L284 ?

Reply:
yeah this is totally my bad.  Fixed and updated class comment